### ISSUE TITLE
keras.ops.quantile returns incorrect symbolic shape for Keras Input with multiple q and axes

### ISSUE THREAD

**Issue by AGFACBNNR** [keras-team/keras#22530]
## Summary
`keras.ops.quantile()` computes the correct value shape for eager tensors, but returns an incorrect symbolic shape when called on a `keras.Input` tensor.

## Reproduction
```python
import keras
import tensorflow as tf

x = tf.random.normal((5, 4, 3))
q = [0.0, 0.5, 1.0]
axis = [1, 2]

print(keras.ops.quantile(x, q=q, axis=axis, method="midpoint").shape)
print(keras.ops.quantile(keras.Input(shape=(4, 3)), q=q, axis=axis, method="midpoint").shape)
```

## Expected
Both calls should infer the same output rank/shape pattern.
For `q` with 3 values and `axis=[1, 2]`, the symbolic shape should be `(3, None)` for an input shaped `(None, 4, 3)`.

## Actual
- Eager tensor result shape: `(3, 5)`
- `keras.Input` result shape: `(None,)`

**Comment by divyashreepathihalli** [comment]
@keras-team /triage
